packages/api/bin/serveUtils.js

```diff
@@ -1,72 +1,183 @@
+      await indexer.indexGranule(es.client, dynamoRecord, es.index);
+      const dbRecord = await translateApiGranuleToPostgresGranule(dynamoRecord, knex);
+      const granulePgModel = new GranulePgModel();
+      await granulePgModel.create(knex, dbRecord);
+    })
   );
 }
 
 async function addProviders(providers) {
+  const knex = await getKnexClient({
+    env: {
+      ...envParams,
+      ...localStackConnectionEnv,
+    },
+  });
+
   const providerModel = new models.Provider();
   const es = await getESClientAndIndex();
   return await Promise.all(
-    providers.map((p) =>
-      providerModel
-        .create(p)
-        .then((provider) => indexer.indexProvider(es.client, provider, es.index)))
+    providers.map(async (p) => {
+      const dynamoRecord = await providerModel.create(p);
+      await indexer.indexProvider(es.client, dynamoRecord, es.index);
+      const dbRecord = await translateApiProviderToPostgresProvider(dynamoRecord);
+      const providerPgModel = new ProviderPgModel();
+      await providerPgModel.create(knex, dbRecord);
+    })
   );
 }
 
 async function addRules(rules) {
+  const knex = await getKnexClient({
+    env: {
+      ...envParams,
+      ...localStackConnectionEnv,
+    },
+  });
+
   const ruleModel = new models.Rule();
   const es = await getESClientAndIndex();
   return await Promise.all(
-    rules.map((r) =>
-      ruleModel
-        .create(r)
-        .then((rule) => indexer.indexRule(es.client, rule, es.index)))
+    rules.map(async (r) => {
+      const dynamoRecord = await ruleModel.create(r);
+      await indexer.indexRule(es.client, dynamoRecord, es.index);
+      const dbRecord = await translateApiRuleToPostgresRule(dynamoRecord, knex);
+      const rulePgModel = new RulePgModel();
+      await rulePgModel.create(knex, dbRecord);
+    })
   );
 }
 
 async function addExecutions(executions) {
+  const knex = await getKnexClient({
+    env: {
+      ...envParams,
+      ...localStackConnectionEnv,
+    },
+  });
+
   const executionModel = new models.Execution();
   const es = await getESClientAndIndex();
-  return await Promise.all(
-    executions.map((e) =>
-      executionModel
-        .create(e)
-        .then((execution) =>
-          indexer.indexExecution(es.client, execution, es.index)))
-  );
+
+  // Since executions has a parent/child relationship with itself
+  // a fake promise is used with reduce() to force records to be created
+  // synchronously
+  const starterPromise = Promise.resolve(null);
+  return await executions.reduce((p, e) => p
+    .then(async () => await executionModel.create(e))
+    .then((dynamoRecord) => {
+      indexer.indexExecution(es.client, dynamoRecord, es.index);
+      return translateApiExecutionToPostgresExecution(dynamoRecord, knex)
+        .then(async (dbRecord) => {
+          const executionPgModel = new ExecutionPgModel();
+          return await executionPgModel.create(knex, dbRecord);
+        });
+    }), starterPromise);
```

Comment:
`Executions` is a little different since it has a parent/child relationship with itself records cannot be asynchronously created without having some records fail to be inserted sometimes. There may be a more elegant way to accomplish this, but currently am using a starter promise with `reduce()` to force records to be inserted in the order they are provided in the `executionsFixture.json` seed file

Reply:
Can you point me to the fixture that has the parent/child executions? I'm not totally following

Reply:
Sure. [Here ya go.](https://github.com/nasa/cumulus-dashboard/blob/CUMULUS-2459/cypress/fixtures/seeds/executionsFixture.json#L105). For dynamo, the order of the records didn't matter, but for Postgres, with the foreign key (parent_cumulus_id) on executions the insert order matters. Since our fixture creates some of these relationships, and I was looking to avoid making changes to existing seed data.

Reply:
So even with this approach, we are relying on the order of the data in the fixture in order to make sure that we insert the parent records before the child records, correct?

If so, that seems a little brittle. I think it might be simpler/more reliable to first `.filter()` the executions from the fixture for records that have no `parentArn` (so possible parents) and create those via Promise.all. Then `filter()` the executions which have `parentArn` and create those

Reply:
That's right. Fair enough. I can add that. Is it possible for a child execution to also be a parent? Schema-wise it's possible. I'm just not sure if it's possible from the business logic side. If that's the case then this change will reduce brittleness, but not remove it entirely. If that scenario is possible/likely then I feel like I'm in the position of having to determine all the dependencies and ensure records are inserted in the correct order.

Reply:
The relationship of executions does have an execution that is both a parent and a child. Splitting the executions into "has parent" and "has no parent" still resulted in records being inserted before their parent occasionally, but unpredictably. I ended up adding some logic to sort the array based on the following rules:
1. if two executions have no parents then they are considered equal and placed anywhere in the array
2. if  execution a has no parent and execution b does have a parent then a must go before b
3. if execution a's arn is equal to execution b's parentArn then a must go before b
4. else a can go after b

I still forced the inserts to run synchronously, but not it doesn't matter how the executions are ordered in the fixture.

Reply:
Got it. Great explanation. Thanks Anthony